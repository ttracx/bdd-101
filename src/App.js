import React from "react";
import "./App.css";
import { Container } from "react-bootstrap";
import { Formik, Form, Field, ErrorMessage } from "formik";
/** @jsx jsx */
import { jsx } from "@emotion/core";
import posed from "react-pose"

export class SignupForm extends React.Component {
  render() {
    return (
      <Formik
        initialValues={{ email: "", password: "" }}
        validate={values => {
          let errors = {};
          if (!values.email) {
            errors.email = "Required";
          } else if (
            !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
          ) {
            errors.email = "Invalid email address";
          }
          return errors;
        }}
        onSubmit={(values, { setSubmitting }) => {
          setTimeout(() => {
            alert(JSON.stringify(values, null, 2));
            setSubmitting(false);
          }, 1000);
        }}
      >
        {({ submitting, errors }) => (
          <Form>
            <Field type="email" name="email" />
            <ErrorMessage name="email" component="div" />
            <button type="submit" disabled={submitting}>
              Submit
            </button>
          </Form>
        )}
      </Formik>

    );
  }
}

class App extends React.Component {
  state = {
    submitted: false,
    confirmation: "Thank you for your submission"
  };
  render() {
    return (
      <Container
        css={{
          color: "hotpink",
          height: "100vh",
          margin: "0 auto",
          display: "flex",
          justifyContent: "center",
          flex: 1,
          alignItems: "center"
        }}
      >
        <SignupForm
          css={{
            alignSelf: "center",
            justifyItem: "center"
          }}
          submitting={this.state.submitting}
          submit={() => {
            this.setState({ submitting: true });
          }}
        />

        <div id="output">
          {this.state.submitting ? this.state.confirmation : ""}
        </div>
      </Container>
    );
  }
}

export default App;
