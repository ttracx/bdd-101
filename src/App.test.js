import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import {Formik} from "formik"
import {SignupForm} from "./App"
Enzyme.configure({ adapter: new Adapter() });

describe('<App />', () => {

  it('renders Home Page', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
    const wrapper = shallow(<App />);

    expect(wrapper).toHaveLength(1);
    expect(renderer
      .create(wrapper)
      .toJSON()).toMatchSnapshot();
  });

  it('should render debug output', ()=>{
    const wrapper = shallow(<App />);
    expect(wrapper.find('#output')).toHaveLength(1)
  })

  it('should have error message field', ()=>{
    const wrapper = shallow(<SignupForm />);
    const test = wrapper.find(Formik).props().validate({email : "test"});
    expect(test).toEqual({email: "Invalid email address"})
    expect(test).toMatchSnapshot()
  })

});