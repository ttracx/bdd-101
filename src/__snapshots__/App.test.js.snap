// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders Home Page 1`] = `
<div
  className="css-y4ox1m container"
>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="email"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    id="output"
  >
    
  </div>
</div>
`;

exports[`<App /> should have error message field 1`] = `
Object {
  "email": "Invalid email address",
}
`;
